Fix failing non-existing author email

hsyn opened the pull request:
Azure deploy user and Collection Build Service can commit to repository without having an email address. `last_azure_dependabot_commit_message` and `recent_azure_commit_messages` methods are failing with key not found error if repository has one of those commits. This commit fixes the issue

hsyn commented:
I think that the build is failing for something that is unrelated to my commit. Let me know If I can be of any help otherwise :)

jurre commented:
This change looks good to me @hsyn, thanks! Could you add a small test for this so we ensure that this feature does not get lost in a refactor/other change in the future?

Regarding the failing tests, those should be fixed if you rebase/merge with master

hsyn commented:
Hi @jurre, thanks for taking your time to take a look at it. I updated the PR with the test as you mentioned. 
I also rebased the branch with latest master. Some other unrelated test is failing now, but you can let me know if I need to rebase/merge again to fix it. 
Have a great day!

jurre commented:
🎉 thanks @hsyn looks good, I've rerun the tests, I think this is just a flake

hsyn commented:
I am glad I could contribute! 🎈
